### ISSUE TITLE
[CoE Starter Kit - QUESTION] current setup for deleted apps

### ISSUE THREAD

**Issue by lingutla123** [microsoft/coe-starter-kit#8631]
### Does this question already exist in our backlog?

- [X] I have checked and confirm this is a new question.

### What is your question?

Hi Jenefer,

Where can we see the setup of deleted apps and flows in COE?

verified in Environment variables couldn't find anything.

Thank you,

### What solution are you experiencing the issue with?

None

### What solution version are you using?

March  2024

### What method are you using to get inventory and telemetry?

None

AB#3792

**Comment by Jenefer-Monroe** [comment]
Hello, please see below for an explanation of how we managed deleted objects in the CoE inventory

### Deleting objects from inventory
**How it works**
When objects are deleted from the tenant, they get marked deleted in the inventory when the following flow is run, along with its helpers: **_CLEANUP - Admin | Sync Template v4 (Check Deleted)_**
This is a long running flow and so it only runs once per week. 

Objects will stay in this state, marked as deleted from inventory, for ~2 months. This is to allow the record to be restored with its metadata in the event the object itself is restored in the tenant. 

At that point, once restore is no longer possible in the tenant, then the records themselves are deleted from the inventory tables as well. 

While they are in the "marked deleted" state they can be viewed in the kit using selecting that view for all the objects as shown here for apps.
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/42ae9af2-6bb5-49fe-afba-838d5f925c68)

**Why keep them around for two months?** 
Here is the scenario we are supporting by keeping these around

- Flow X created in tenant
- Flow X added to the CoE inventory tables
- Flow X marked as exempt from inactivity processing in the CoE
- Flow X deleted from tenant
- Flow X marked as deleted in CoE inventory tables
- Flow X restored in tenant
- Flow X unmarked deleted in CoE inventory tables, exemption from inactivity processing remains in place.

**I need these records to stay in perpetuity**
We have an env var, _Also Delete from CoE_, which allows you to prevent the deletion from the inventory tables. Default behavior (yes) described above, but if you set it to no then these records will never be deleted by the kit

![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/af05bf15-e937-4032-9eab-e4d290108147)


**I don't want these records sitting around at all**
If you want the deletion from inventory to occur immediately, or at some other cadence like 6 months, then you will need to create your own flow on the side to do this. You can use the mechanism used in _CLEANUP - Admin | Sync Template v4 (Check Deleted)_ as a template for how to do this.
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/187ec89f-77d7-48d4-8fb3-ca6df9b704d5)

**Comment by lingutla123** [comment]
Thank you Jenefer.

**Comment by lingutla123** [comment]
Hi Jenefer,

How to restore the deleted Apps and Flows in 'Power Platform Admin view' app?

What if i created new flow for custom time to delete the apps and flows and turn off the existing CLEANUP - Admin | Sync Template v4 (Check Deleted) flow. Is CLEANUP - Admin | Sync Template v4 (Check Deleted) flow remains turned off after installing the monthly updates?

Thank you,

**Comment by Jenefer-Monroe** [comment]
You cannot restore objects from within the kit, you need to go to through product features to do that. We just ensure that the metadata (ex excuse from x) will remain after restore. 
Here is a blog someone wrote with details: https://community.dynamics.com/blogs/post/?postid=6cbb3914-1ee1-423d-8ac4-156e97cf58df

**Comment by Jenefer-Monroe** [comment]
Yes upgrade should not turn on any flows by default other than the couple we ship turned on (related to setup wizard) 
So anything you turn off should remain off on upgrades.

**Comment by harini-2-y** [comment]
We haven’t heard back in a while, so we’ll proceed to close this issue for now. If the issue persists or you need further assistance, feel free to log a new issue. We’re happy to help anytime. Thanks again for your engagement!

**Comment by harini-2-y** [comment]
We noticed that this issue has been pending for quite some time, and we have not received any recent updates from your end. In the absence of further communication, we are proceeding to close this ticket, assuming that your concerns have been addressed or resolved. 
If you continue to experience any difficulties related to this matter, please feel free to open a new support ticket. This will help us ensure prompt and effective assistance, rather than reopening the new Issue. 
Thank you for your understanding and cooperation.
